src/input/type_record.cc

```diff
@@ -265,6 +268,9 @@ bool Record::finish()
     data_->finished = true;
     for (vector<Key>::iterator it=data_->keys.begin(); it!=data_->keys.end(); it++)
     {
+        ASSERT(typeid( *(it->type_) ) != typeid(Parameter), "Finished Record '%s' can't contain key '%s' of type Parameter.\n",
```

Comment:
1) Finish of generic types has to be called when they are instantiated.
2) Finish of generic types is an error when called during pass through IST.
Thus we want to distinguish two different favour of finish. This can be done by optional parameter generic_allowed=false.

When called during instantiation we call finish(true), otherwise generic is no allowed. 

This way we doesn't need generic_ flag. Just use this ASSERT only in case generic_allowed == false.